### ISSUE TITLE
Fabric server-side mods support

### ISSUE THREAD

**Issue by JustFoxx** [AnchorMC/Anchor#1]
It would be better to make a fabric serverside mods to be supported than making plugins support or the best would be support mods and plugins at the same time but it could have some problems

**Comment by ErrorNoInternet** [comment]
Fabric mods are written in Java, and I'm not sure if its possible to load Java code (let alone Fabric mods that rely on the Fabric API) into a pure Go project...

**Comment by JustFoxx** [comment]
there are many JVM compilers for Go

**Comment by MrMelon54** [comment]
maybe just write a java wrapper for the fabric api.. then hook that into the go server..
unfortunately mixins would be really difficult to get working..
and would probably require specific support per mod..
but mods that add items to the game would work...
a plugin system is probably easier to setup though...

**Comment by PassTheMayo** [comment]
Late reply, but I do not think having a JVM-Go interface would be very optimized, especially for any mods that are resource heavy or do a lot of data processing on the server-side. I don't exactly have a plan for how plugins/mods will be implemented, but I am thinking that Lua may be a good choice since it is lightweight, highly efficient and simple to write. I would like to hear everybody's opinion on this but this may not be implemented for quite some time.

**Comment by MrMelon54** [comment]
Yeah lua is probably a better option, maybe plugins in Go as well?

**Comment by PassTheMayo** [comment]
> Yeah lua is probably a better option, maybe plugins in Go as well?

Yeah I think so as well. I'll plan on supporting both Lua plugins as well as Go plugins that can be built in with the server software by adding custom source code.

**Comment by jLn0n** [comment]
maybe use [luau](https://luau-lang.org) instead? it is more optimized but i don't know if there are any ports that are written in go
